src/Table/CheckboxTable.component.tsx

```diff
@@ -0,0 +1,318 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import ReactTable from 'react-table';
+import TablePagination from '../TablePagination/TablePagination.component';
+import 'react-table/react-table.css';
+
+import selectTableHOC from 'react-table/lib/hoc/selectTable';
+
+import { Icon } from 'react-icons-kit';
+import { bars as IconLarge } from 'react-icons-kit/fa/bars';
+import { ic_format_align_justify as IconCondensed }
+  from 'react-icons-kit/md/ic_format_align_justify';
+import { ic_view_headline as IconMedium }
+  from 'react-icons-kit/md/ic_view_headline';
+import { threeHorizontal as IconMore } from 'react-icons-kit/entypo/threeHorizontal';
+import Popover from '../Popover/Popover.component';
+
+type Props = {
```

Comment:
We need to allow the user to pass in any props available to the underlying component. You would ```
import TableProps props from 'react-table';```
and then: ```
type Props = Partial<TableProps> & {
   ...theRestofThePropsYouNeed
};```